app/models/observation.rb

```diff
@@ -193,6 +238,183 @@ class Observation < AbstractModel
+    end
+  }
+  scope :of_name_like,
+        ->(name) { where(name: Name.text_name_includes(name)) }
+  scope :by_user,
+        ->(user) { where(user: user) }
+  scope :with_location,
+        -> { where.not(location: nil) }
+  scope :without_location,
+        -> { where(location: nil) }
+  scope :at_location,
+        ->(location) { where(location: location) }
+  scope :in_region,
+        ->(where) { where(Observation[:where].matches("%#{where}")) }
+  scope :in_box, # Use named parameters (n, s, e, w), any order
+        lambda { |**args|
+          if args[:s]&.between?(0, 90) && args[:n]&.between?(0, 90) &&
+             args[:w]&.between?(-180, 180) && args[:e]&.between?(-180, 180) &&
+             args[:s] <= args[:n] &&
+             ((args[:w] <= args[:e]) || (args[:w] >= 0 && args[:e] <= 0))
+
+            # expand box by epsilon to create leeway for Float rounding
+            # Fixes a bug where Califoria fixture was not in a box
+            # defined by the fixture's north, south, east, west
+            epsilon = 0.00001
+            n = args[:n] + epsilon
+            s = args[:s] - epsilon
+            e = args[:e] + epsilon
+            w = args[:w] - epsilon
+
+            # Does box straddle 180 deg?
+            if args[:e] < args[:w]
+              where(
+                # Observation[:long] between w & 180 OR between 180 and e
+                (Observation[:lat] >= s).
+                and(Observation[:lat] <= n).
+                and(Observation[:long] >= w).
+                or(Observation[:long] <= e)
+              )
+            else
+              where(
+                (Observation[:lat] >= s).
+                and(Observation[:lat] <= n).
+                and(Observation[:long] >= w).
+                and(Observation[:long] <= e)
+              )
+            end
+          else
+            Observation.none
+          end
+        }
+  scope :is_collection_location,
+        -> { where(is_collection_location: true) }
+  scope :not_collection_location,
+        -> { where(is_collection_location: false) }
+  scope :with_image,
+        -> { where.not(thumb_image: nil) }
+  scope :without_image,
+        -> { where(thumb_image: nil) }
+  scope :with_notes,
+        -> { where.not(notes: Observation.no_notes) }
+  scope :without_notes,
+        -> { where(notes: Observation.no_notes) }
+  scope :has_notes_field,
+        ->(field) { where(Observation[:notes].matches("%:#{field}:%")) }
+  scope :notes_include,
+        ->(notes) { where(Observation[:notes].matches("%#{notes}%")) }
+  scope :with_specimen,
+        -> { where(specimen: true) }
+  scope :without_specimen,
+        -> { where(specimen: false) }
+  scope :with_sequence,
+        -> { joins(:sequences).distinct }
+  scope :without_sequence,
+        -> { missing(:sequences) }
+  scope :confidence, lambda { |min, max = min| # confidence between min & max %
+    where(vote_cache: (min.to_f / (100 / 3))..(max.to_f / (100 / 3)))
+  }
```

Comment:
@nimmolo: Is this OK? It's easy enough to change/expand if that would work better for GraphQL.
At your convenience, please let me know how you'd like to proceed.

I decided to do this instead of `has_confidence_at_least`, and `has_confidence_less_than` because:
- It's like the existing UI for pattern searches
- Percents are easier to understand than vote_cache
- You can use it to get confidence >= n (`.confidence(n, 100)`) or < n (`where.not(Observation.confidence(n, 100)`
- The name is shorter
- Rubocop doesn't like predicate methods that start with `has_`.
https://docs.rubocop.org/rubocop/cops_naming.html#namingpredicatename
So I try to avoid them altogether to avoid confusion. And this is vaguely similar to a predicate.

Some options to think about:
- Scrap this, and do exactly what you asked for;
- Rename this `with_confidence_between`
- Add convenience scopes `with_confidence_at_least` `with_confidence_less_than`